merkletree/pad.go

```diff
@@ -76,48 +93,65 @@ func (pad *PAD) Update(policies Policies) {
 		pad.loadedEpochs = append(pad.loadedEpochs[:0], pad.loadedEpochs[n:]...)
 	}
 
-	pad.updateInternal(policies, pad.currentSTR.epoch+1)
+	pad.updateInternal(policies, pad.latestSTR.epoch+1)
 }
 
 func (pad *PAD) Set(key string, value []byte) error {
-	return pad.tree.Set(key, value)
+	str := pad.latestSTR
+	if str == nil {
+		return ErrorSTRNotFound
+	}
+	index, _ := pad.computePrivateIndexInEpoch(key, str.epoch)
+	return pad.tree.Set(index, key, value)
 }
 
-func (pad *PAD) Lookup(key string) *AuthenticationPath {
-	str := pad.currentSTR
-	return str.tree.Get(key)
+func (pad *PAD) Lookup(key string) (*AuthenticationPath, error) {
+	return pad.LookupInEpoch(key, pad.latestSTR.epoch)
 }
 
 func (pad *PAD) LookupInEpoch(key string, epoch uint64) (*AuthenticationPath, error) {
 	str := pad.GetSTR(epoch)
 	if str == nil {
 		return nil, ErrorSTRNotFound
 	}
-	ap := str.tree.Get(key)
+	lookupIndex, proof:= pad.computePrivateIndexInEpoch(key, str.epoch)
+	ap := str.tree.Get(lookupIndex)
+	ap.vrfProof = proof
 	return ap, nil
 }
 
 func (pad *PAD) GetSTR(epoch uint64) *SignedTreeRoot {
-	if epoch >= pad.currentSTR.epoch {
-		return pad.currentSTR
+	if epoch >= pad.latestSTR.epoch {
+		return pad.latestSTR
 	}
 	return pad.snapshots[epoch]
 }
 
 func (pad *PAD) TB(key string, value []byte) (*TemporaryBinding, error) {
-	//FIXME: compute private index twice
-	//it would be refactored after merging VRF integration branch
-	index := computePrivateIndex(key)
-	tb := pad.currentSTR.sig
+	str := pad.latestSTR
+	if str == nil {
```

Comment:
Hmm, this makes me think that the `seed` should be generated at `PAD` construction time, since it's needed right away for the temporary bindings.